Generator/Sources/NeedleFramework/Generating/Pluginized/PluginizedDependencyProviderContentTask.swift

```diff
@@ -0,0 +1,143 @@
+
+        nonCoreComponentNames = Set(pluginizedComponents.map { pluginizedComponent in
+            pluginizedComponent.nonCoreComponent.name
+        })
+    }
+
+    /// Execute the task and returns the processed in-memory dependency graph
+    /// data models.
+    ///
+    /// - returns: The list of `ProcessedDependencyProvider`.
+    override func execute() -> [PluginizedProcessedDependencyProvider] {
+
+        return providers.map { (provider: DependencyProvider) -> PluginizedProcessedDependencyProvider in
+            do {
+                if provider.pathContains(anyOf: nonCoreComponentNames) {
+                    return try process(provider, auxillaryPropertyMap: nonCoreComponentMap)
+                } else {
+                    return try process(provider, auxillaryPropertyMap: pluginExtensionMap)
+                }
+            } catch DependencyProviderContentError.propertyNotFound(let info) {
+                var message = "Could not find a provider for \(info.name): \(info.type) which was required by \(info.dependency)."
+                if let possibleMatchComponent = info.possibleMatchComponent {
+                    message += " Found possible matches \(info.possibleNames) at \(possibleMatchComponent)."
+                }
+                fatalError(message)
+            } catch {
+                fatalError("Unhandled error while processing dependency provider content: \(error)")
+            }
+        }
+    }
+
+    // MARK: - Private
+
+    struct AuxillaryProperties {
+        let sourceName: String
+        let properties: [Property]
+    }
+
+    private let providers: [DependencyProvider]
+    private let nonCoreComponentMap: [String: AuxillaryProperties]
+    private let pluginExtensionMap: [String: AuxillaryProperties]
+    private let nonCoreComponentNames: Set<String>
+
+    private func process(_ provider: DependencyProvider, auxillaryPropertyMap: [String: AuxillaryProperties]) throws -> PluginizedProcessedDependencyProvider  {
+        var levelMap = [String: Int]()
+
+        let properties = try provider.dependency.properties.map { (property : Property) -> PluginizedProcessedProperty in
+            var level = 0
+            let revesedPath = provider.path.reversed()
+            for component in revesedPath {
+                if component.properties.contains(property) {
+                    levelMap[component.name] = level
+                    return PluginizedProcessedProperty(data: ProcessedProperty(unprocessed: property, sourceComponentType: component.name), auxillarySourceType: nil)
+                } else if let auxillaryProperties = auxillaryPropertyMap[component.name] {
+                    if auxillaryProperties.properties.contains(property) {
+                        levelMap[component.name] = level
+                        return PluginizedProcessedProperty(data: ProcessedProperty(unprocessed: property, sourceComponentType: auxillaryProperties.sourceName), auxillarySourceType: .pluginExtension)
```

Comment:
The `auxillarySourceType` should be passed into this method right? It's not always `pluginExtension`.

Reply:
done